### PR TITLE
add test and publish

### PR DESCRIPTION
**Pull request by davydyuk**
Signed-off-by: Vitalii Davydiuk <ext-vitalii.davydiuk@here.com>

### CODE REVIEW

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -1,9 +1,42 @@
...
   - veracode scan
   - deps scan
   - generate pages
 
 
+test:
+  stage: test
+  tags:
+    - docker-sock
+  image: advancedtelematic/gitlab-jobs:0.2.0
```
Comment (simao) [inline .gitlab-ci.yml]: can use 0.2.3

File: .gitlab-ci.yml [hunk 2]
```diff
@@ -1,9 +1,42 @@
...
+  before_script:
+    - docker ps
+    - ./deploy/cleanup.sh
+    - ./deploy/ci_setup.sh
+  script:
+    - docker ps
+    - ./sbt -sbt-dir ./.sbt -ivy ./.ivy2 clean test
```
Comment (simao) [inline .gitlab-ci.yml]: use `sbt`

File: .gitlab-ci.yml [hunk 3]
```diff
@@ -1,9 +1,42 @@
...
+
+publish:
+  stage: publish
+  tags:
+    - docker-sock
+  image: advancedtelematic/gitlab-jobs:0.2.0
+#  only:
+#    - master
+  script:
+    - ./sbt -x -sbt-dir ./.sbt -ivy ./.ivy2 +publish
```
Comment (simao) [inline .gitlab-ci.yml]: same

Comment (simao) [reply .gitlab-ci.yml]: to publish see ota-tuf which already publishes some jars to nexus

File: .gitlab-ci.yml (hunk 1 above)
Comment (davydyuk) [reply .gitlab-ci.yml]: done

**Comment by davydyuk** [comment]
@zabbal, done

File: .gitlab-ci.yml (hunk 2 above)
Comment (davydyuk) [reply .gitlab-ci.yml]: done

File: .gitlab-ci.yml (hunk 3 above)
Comment (davydyuk) [reply .gitlab-ci.yml]: done

**Comment by simao** [comment]
Is there a successful pipeline of this we can look at?

**Comment by davydyuk** [comment]
@simao https://main.gitlab.in.here.com/olp/edge/ota/connect/back-end/libats/pipelines/180470
Artifacts are published:
http://nexus.advancedtelematic.com:8081/content/repositories/releases/com/advancedtelematic/
to publish them I have commented https://github.com/advancedtelematic/libats/pull/130/files#diff-96edf7a6f008de9e928d04e1ae5e12a5R43-R44
Please check this MR and I'll delete the comments

File: .gitlab-ci.yml [hunk 4]
```diff
@@ -1,8 +1,60 @@
...
+#    - master
+  except:
+    - schedules
+  script:
+    - git fetch origin
+    - git describe
+    - export VAULT_TOKEN=$(cat /opt/vault/token)
+    - gitlab-docker-login $VAULT_ADDR
+    - eval $(set-vault-env -v $VAULT_ADDR --secret gitlab/ats-aws-key --secret gitlab/maven-publish)
+    - sbt -x -sbt-dir ./.sbt -ivy ./.ivy2 +publish
```
Comment (simao) [inline .gitlab-ci.yml]: Please remove `-x`

**Pull request merged**